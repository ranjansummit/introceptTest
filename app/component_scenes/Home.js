import React, {Component} from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  Keyboard,
  TextInput,
  Image,
  TouchableOpacity,
  StatusBar,
  PixelRatio,
  Alert,
} from 'react-native';

import {
  Table,
  TableWrapper,
  Row,
  Rows,
  Col,
  Cols,
  Cell,
} from 'react-native-table-component';
import {connect} from 'react-redux';
import CountryPicker from 'react-native-country-picker-modal';
import Theme from '../uielements/Utility/Colors';
class Home extends Component {
  constructor(props) {
    super(props);
    const elementButton = value => (
      <TouchableOpacity onPress={() => this._alertIndex(value)}>
        <View style={styles.btn}>
          <Text style={styles.btnText}>{value}</Text>
        </View>
      </TouchableOpacity>
    );

    this.state = {
      name: '',
      email: '',
      tableHead: [
        elementButton('Name'),
        elementButton('Country'),
        elementButton('Mobile Number'),
        elementButton('Mobile Brand'),
      ],
      tableTitle: [
        'Title',
        'Title2',
        'Title3',
        'Title4',
        'Title',
        'Title2',
        'Title3',
        'Title4',
        'Title',
        'Title2',
        'Title3',
        'Title4',
        'Title',
        'Title2',
        'Title3',
        'Title4',
        'Title',
        'Title2',
        'Title3',
        'Title4',
        'Title',
        'Title2',
        'Title3',
        'Title4',
      ],
      tableData: [
        [, 'a', 'b', 'c', 'd'],
        ['1', '2', '3', '4'],
        ['a', 'b', 'c', 'd'],
      ],
      isCountryclicked: false,
      value: [
        {key: '1', width: 33, title: 'name'},
        {key: '2', width: 33, title: 'mobileno', sortable: true},
        {key: '3', width: 33, title: 'mobilebrand', sortable: true},
      ],
      field: [
        {name: 'ranjan', mobileno: '9840051137', mobilebrand: 'nokia'},
        {name: 'ranjan', mobileno: '9840051137', mobilebrand: 'nokia'},
        {name: 'ranjan', mobileno: '9840051137', mobilebrand: 'nokia'},
      ],
      mobile_number: '',
      password: '',
      confirm_password: '',
      checked: false,
      error1: false,
      error2: false,
      error3: false,
      error4: false,
      error5: false,
      password_error: '',
      cca2: 'US',
      confirm_password_error: '',
      country: '',
    };
  }
  _alertIndex(value) {
    Alert.alert(`This is column ${value}`);
  }

  render() {
    return (
      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
        <ScrollView style={{marginTop: 30}}>
          {this.props.loading ? <Spinner /> : null}
          {this.props.error ? (
            this.props.error.email ? (
              <View>{this.emailAlert()}</View>
            ) : this.props.error.mobile ? (
              <View>{this.mobileAlert()}</View>
            ) : null
          ) : null}
          {this.props.reg_success === false ? (
            <View>{this.successAlert()}</View>
          ) : null}
          <View style={styles.container}>
            <View>
              <View>
                <Text style={styles.bigtext}>Introcept Survey</Text>
              </View>
            </View>
            <View>
              <View>
                <TextInput
                  style={styles.inputText}
                  type="string"
                  required
                  placeholder="Full Name"
                  placeholderTextColor={Theme.colors.navyBlue}
                  onChangeText={name => this.setState({name, error1: false})}
                  value={this.state.name}
                />
              </View>
              <View style={styles.error}>
                {this.state.error1 ? <ErrorIcon /> : null}
              </View>
            </View>
            {this.state.error1 ? (
              <View style={{justifyContent: 'flex-end', marginRight: 5}}>
                <Text
                  style={{
                    color: 'white',
                    fontSize: 0.04 * screenWidth,
                    textAlign: 'center',
                    padding: 2,
                    marginTop: 2,
                    marginBottom: 2,
                    marginLeft: '30%',
                    backgroundColor: 'black',
                    borderTopWidth: 5,
                    borderTopColor: 'red',
                  }}>
                  Please provide your name
                </Text>
              </View>
            ) : null}

            <View>
              <View
                style={{
                  height: 45,
                  marginTop: 15,
                  marginLeft: 5,
                  marginRight: 5,
                  flexDirection: 'row',
                }}>
                <View
                  style={{
                    marginRight: 3,
                    flex: 0.4,
                    backgroundColor: 'white',
                    flexDirection: 'row',
                    padding: 10,

                    alignItems: 'center',
                  }}>
                  <TextInput
                    autoCapitalize="none"
                    style={{
                      height: 45,
                      backgroundColor: 'white',
                      padding: 10,
                      fontSize: 18,
                      color: Theme.colors.navyBlue,
                    }}
                    type="string"
                    required
                    placeholder="Country"
                    placeholderTextColor={Theme.colors.navyBlue}
                    onChangeText={mobile_number =>
                      this.setState({mobile_number, error3: false})
                    }
                    value={this.state.mobile_number}
                  />
                </View>
                <View
                  style={{
                    flex: 0.6,
                    backgroundColor: 'red',
                    justifyContent: 'flex-start',
                  }}>
                  <TextInput
                    autoCapitalize="none"
                    style={{
                      height: 45,
                      backgroundColor: 'white',
                      padding: 10,
                      fontSize: 18,
                      color: Theme.colors.navyBlue,
                    }}
                    type="string"
                    keyboardType="numeric"
                    required
                    placeholder="Mobile Number"
                    placeholderTextColor={Theme.colors.navyBlue}
                    onChangeText={mobile_number =>
                      this.setState({mobile_number, error3: false})
                    }
                    value={this.state.mobile_number}
                  />
                </View>
              </View>
              <View style={styles.error}>
                {this.state.error3 ? <ErrorIcon /> : null}
              </View>
            </View>
            {this.state.error3 ? (
              <View style={{justifyContent: 'flex-end', marginRight: 5}}>
                <Text
                  style={{
                    color: 'white',
                    fontSize: 0.04 * screenWidth,
                    textAlign: 'center',
                    padding: 2,
                    marginBottom: 2,
                    marginTop: 2,

                    marginLeft: '30%',
                    backgroundColor: 'black',
                    borderTopWidth: 4,
                    borderTopColor: 'red',
                  }}>
                  Invalid mobile number.
                </Text>
              </View>
            ) : null}
            <View>
              <View>
                <TextInput
                  autoCapitalize="none"
                  style={styles.inputText}
                  type="string"
                  required
                  placeholder="Brand of the Phone"
                  placeholderTextColor={Theme.colors.navyBlue}
                  onChangeText={password =>
                    this.setState({password, error4: false})
                  }
                  value={this.state.password}
                />
              </View>
              <View style={styles.error}>
                {this.state.error4 ? <ErrorIcon /> : null}
              </View>
            </View>
            {this.state.error4 ? (
              <View style={{justifyContent: 'flex-end', marginRight: 5}}>
                <Text
                  style={{
                    color: 'white',
                    fontSize: 0.04 * screenWidth,
                    textAlign: 'center',
                    padding: 2,
                    marginTop: 2,
                    marginBottom: 2,
                    marginLeft: '30%',
                    backgroundColor: 'black',
                    borderTopWidth: 4,
                    borderTopColor: 'red',
                  }}>
                  {this.state.password_error}
                </Text>
              </View>
            ) : null}
            <View
              style={{
                flexDirection: 'row',
                marginTop: 5,
                marginBottom: 10,
              }}></View>
            <TouchableOpacity
              style={styles.button}
              onPress={() =>
                this.onRegister(
                  this.state.name,
                  this.state.email,
                  this.state.mobile_number,
                  this.state.password,
                  this.state.confirm_password,
                  this.state.checked,
                )
              }>
              <Text style={styles.text}>Save your Detail</Text>
            </TouchableOpacity>
            <Text style={styles.bigtextdata}> Survey Data</Text>

            <Table borderStyle={{borderWidth: 4, borderColor: '#c8e1ff'}}>
              <Row
                data={this.state.tableHead}
                style={styles.head}
                textStyle={styles.tabletext}
              />
              <Rows data={this.state.tableData} textStyle={styles.tabletext} />
            </Table>

          </View>
        </ScrollView>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    margin: 5,
    padding: 5,
    flex: 1,
    justifyContent: 'space-evenly',
    backgroundColor: Theme.colors.lightBlue,
  },
  inputText: {
    padding: 10,
    height: 45,
    flex: 10,
    fontSize: 18,
    color: Theme.colors.navyBlue,
    backgroundColor: 'white',
    marginTop: 25,
    marginLeft: 5,
    marginRight: 5,
  },
  button: {
    justifyContent: 'center',
    backgroundColor: Theme.colors.red,
    padding: 10,
    margin: 10,
    height: 45,
  },
  text: {
    fontSize: 18,
    color: 'white',
    textAlign: 'center',
  },
  error: {
    position: 'absolute',
    top: 12,
    right: 30,
    bottom: 0,
    justifyContent: 'center',
    alignSelf: 'flex-end',
  },
  bigtext: {
    fontSize: 25,
    color: Theme.colors.navyBlue,
    marginTop: 35,
    textAlign: 'center',
  },
  bigtextdata: {
    fontSize: 25,
    color: Theme.colors.navyBlue,
    marginTop: 15,
    marginBottom: 10,

    textAlign: 'center',
  },
  tablecontainer: {
    flex: 1,
    padding: 16,
    paddingTop: 30,
    backgroundColor: '#fff',
  },
  singleHead: {width: 80, height: 40, backgroundColor: '#c8e1ff'},
  head: {flex: 1, backgroundColor: '#c8e1ff'},
  title: {flex: 2, backgroundColor: '#f6f8fa'},
  titleText: {marginRight: 6, textAlign: 'right'},
  tabletext: {
    textAlign: 'left',
    color: Theme.colors.navyBlue,
    paddingHorizontal: 4,
  },
  tablebtn: {
    width: 58,
    height: 18,
    marginLeft: 15,
    backgroundColor: '#c8e1ff',
    borderRadius: 2,
  },
  tablebtnText: {textAlign: 'center'},
});

const mapStateToProps = state => {
  return {};
};
export default connect()(Home);
