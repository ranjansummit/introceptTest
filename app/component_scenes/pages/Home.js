import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  Keyboard,
  TextInput,
  TouchableOpacity,
  Alert
}
  from 'react-native';

import {
  Table,
  Row,
  Rows
}
  from 'react-native-table-component';
import { connect } from 'react-redux';
import CountryPicker from 'react-native-country-picker-modal';
import Theme from '../../uielements/Utility/Colors';
console.disableYellowBox = true;


 class Home extends Component {
  constructor(props) {
    super(props);
    const elementButton = (value,title) => (
<View style={{flexDirection: 'row', justifyContent: 'center', alignItems: 'center' }}>
        <Text>{title} </Text>
        { value==1|| value ==3 ? (

        <TouchableOpacity onPress={() => this._sortData(value)}>
          <Text style={{color: 'red'}}>sort</Text>
        </TouchableOpacity>
        ): null }
      </View>

    );

    this.state = {
      name: '',
      email: '',
      tableHead: [
        elementButton(0,'Name'),
        elementButton(1,'Country'),
        elementButton(2,'Mobile No.'),
        elementButton(3,'Brand'),
      ],
      tableData: [],
      mobile_number: '',
      brand: '',
      confirm_password: '',
      country: 'Nepal',
      checked: false,
      nameError: false,
      mobileError: false,
      brandError: false,
      callingCode: 'NP',
      countryCode: 'NP',

    };
  }

  _onSaveEntryClicked = (name, number, brand) => {

    if (name.length === 0 || brand.length === 0|| number.length === 0)  {
      Alert.alert("Please enter all values properly");
    }
    else {
      this._SaveData()
    }
  }


  // to save data
  _SaveData() {
    var data = this.state.tableData
    data.push([this.state.name, this.state.country, this.state.mobile_number, this.state.brand]);
    this.setState({ tableData: data })
  }

  //  to sort data
  _sortData(value) {
    var data = this.state.tableData
    data.sort((a, b) => a[value].localeCompare(b[value]));
    this.setState({ tableData: data })
  }
  _pickerOnselect= country => {
    this.setState({
      countryCode: country.cca2,
      callingCode: `+ ${country.callingCode}`,
      country: country.name
    });
  }
  

  render() {
    return (
      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
        <ScrollView style={{ marginTop: 30 }}>

          <View style={styles.container}>
            <View>
              <View>
                <Text style={styles.bigtext}>Introcept Survey</Text>
              </View>
            </View>
            <View>
              <View>
                <TextInput
                  style={styles.inputText}
                  type="string"
                  required
                  placeholder="Full Name"
                  placeholderTextColor={Theme.colors.navyBlue}
                  onChangeText={name => this.setState({ name, error1: false })}
                  value={this.state.name}
                />
              </View>
         
            </View>

            <View>
              <View
                style={{
                  height: 45,
                  marginTop: 15,
                  marginLeft: 5,
                  marginRight: 5,
                  flexDirection: 'row',
                }}>
                <View
                  style={{
                    marginRight: 3,
                    flex: 0.4,
                    backgroundColor: 'white',
                    flexDirection: 'row',
                    padding: 10,

                    alignItems: 'center',
                  }}>
                  <CountryPicker
                    withEmoji={false}
                    withCloseButton={false}
                    withFlag={true}
                    countryCode={this.state.countryCode}
                    withFilter={true}
                    theme="black"
                    withCallingCode
                    onSelect={country => {
                      this.setState({
                        countryCode: country.cca2,
                        callingCode: `+ ${country.callingCode}`,
                        country: country.name
                      });
                    }}
                  />

                  <Text style={styles.pickertext}>{this.state.country}
                 
                  </Text>

                </View>
                <View
                  style={{
                    flex: 0.6,
                    backgroundColor: 'red',
                    justifyContent: 'flex-start',
                  }}>
                  <TextInput
                    autoCapitalize="none"
                    style={{
                      height: 45,
                      backgroundColor: 'white',
                      padding: 10,
                      fontSize: 18,
                      color: Theme.colors.navyBlue,
                    }}
                    type="string"
                    keyboardType="numeric"
                    required
                    placeholder="Mobile Number"
                    placeholderTextColor={Theme.colors.navyBlue}
                    onChangeText={mobile_number =>
                      this.setState({ mobile_number, error3: false })
                    }
                    value={this.state.mobile_number}
                  />
                </View>
              </View>
              </View>
          
           
            <View>
              <View>
                <TextInput
                  autoCapitalize="none"
                  style={styles.inputText}
                  type="string"
                  required
                  placeholder="Brand of the Phone"
                  placeholderTextColor={Theme.colors.navyBlue}
                  onChangeText={brand =>
                    this.setState({ brand, error4: false })
                  }
                  value={this.state.brand}
                />
              </View>
              </View>
            <View
              style={{
                flexDirection: 'row',
                marginTop: 5,
                marginBottom: 10,
              }}></View>
            <TouchableOpacity
              style={styles.button}
              onPress={() =>
                this._onSaveEntryClicked(
                  this.state.name,
                  this.state.mobile_number,
                  this.state.brand
                )
              }>
              <Text style={styles.buttonText}>Save your Detail</Text>
            </TouchableOpacity>
            <Text style={styles.bigtextdata}> Survey Data</Text>
            <Table borderStyle={{ borderWidth: 4, borderColor: '#c8e1ff' }}>
              <Row
                data={this.state.tableHead}
                style={styles.tableHead}
                textStyle={styles.tabletext}
              />
              <Rows data={this.state.tableData} textStyle={styles.tabletext} />
            </Table>

          </View>
        </ScrollView>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    margin: 5,
    padding: 5,
    flex: 1,
    justifyContent: 'space-evenly',
    backgroundColor: Theme.colors.lightBlue,
  },
  inputText: {
    padding: 10,
    height: 45,
    flex: 10,
    fontSize: 18,
    color: Theme.colors.navyBlue,
    backgroundColor: 'white',
    marginTop: 25,
    marginLeft: 5,
    marginRight: 5,
  },
  button: {
    justifyContent: 'center',
    backgroundColor: Theme.colors.red,
    padding: 10,
    margin: 10,
    height: 45,
  },
  buttonText: {
    fontSize: 18,
    color: 'white',
    textAlign: 'center',
  },
  error: {
    position: 'absolute',
    top: 12,
    right: 30,
    bottom: 0,
    justifyContent: 'center',
    alignSelf: 'flex-end',
  },
  bigtext: {
    fontSize: 25,
    color: Theme.colors.navyBlue,
    marginTop: 35,
    textAlign: 'center',
  },
  bigtextdata: {
    fontSize: 25,
    color: Theme.colors.navyBlue,
    marginTop: 15,
    marginBottom: 10,
    textAlign: 'center',
  },
  tablecontainer: {
    flex: 1,
    padding: 16,
    paddingTop: 30,
    backgroundColor: '#fff',
  },
  tableHead: {
    flex: 1,
    backgroundColor: '#c8e1ff'
  },
  title: {
    flex: 2,
    backgroundColor: '#f6f8fa'
  },
  titleText: {
    marginRight: 6,
    textAlign: 'right'
  },
  tabletext: {
    textAlign: 'left',
    color: Theme.colors.navyBlue,
    paddingHorizontal: 4,
  },
  tablebtn: {
    width: 58,
    height: 18,
    marginLeft: 15,
    backgroundColor: '#c8e1ff',
    borderRadius: 2,
  },
  tablebtnText: {
    textAlign: 'center'
  },
  pickertext: {
    paddingTop: 5,
    color: Theme.colors.navyBlue,
    fontSize: 18
  }
});

const mapStateToProps = state => {
  return {};
};
export default connect()(Home);
