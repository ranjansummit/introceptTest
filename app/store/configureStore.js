import {createStore, applyMiddleware, compose} from 'redux';
import reducers from '../component_scenes/reducer';
import createSagaMiddleware from 'redux-saga';
import sagas from '../rootSaga/RootSaga';

export default function configureStore() {
  const sagaMiddleware = createSagaMiddleware();

  let store = createStore(reducers, compose(applyMiddleware(sagaMiddleware)));
  sagaMiddleware.run(sagas);
  return store;
}
