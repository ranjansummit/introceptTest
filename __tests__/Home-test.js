import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Home from '../app/component_scenes/pages/Home';
import { Text } from 'react-native'
import mockComponent from 'react-native/jest/mockComponent'

const mockStore = configureStore([]);
describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    
    });
    component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    );
  });
  //  it is just basically testing if the home  component is rendered correcly or not
  it('redered sucesfully', () => {
    const comp = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    expect(comp).toMatchSnapshot();


  });
});