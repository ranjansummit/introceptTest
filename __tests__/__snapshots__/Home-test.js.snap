// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component redered sucesfully 1`] = `
<RCTScrollView
  accessible={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#E1E5F2",
          "flex": 1,
          "justifyContent": "space-evenly",
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <View>
        <View>
          <Text
            style={
              Object {
                "color": "#012D6C",
                "fontSize": 25,
                "marginTop": 35,
                "textAlign": "center",
              }
            }
          >
            Introcept Survey
          </Text>
        </View>
      </View>
      <View>
        <View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Full Name"
            placeholderTextColor="#012D6C"
            rejectResponderTermination={true}
            required={true}
            style={
              Object {
                "backgroundColor": "white",
                "color": "#012D6C",
                "flex": 10,
                "fontSize": 18,
                "height": 45,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 25,
                "padding": 10,
              }
            }
            type="string"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 45,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 0.4,
                "flexDirection": "row",
                "marginRight": 3,
                "padding": 10,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-start",
                      "justifyContent": "flex-start",
                    },
                    Object {
                      "marginTop": 5,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginRight": 10,
                      "width": 30,
                    }
                  }
                >
                  <ActivityIndicator
                    animating={true}
                    color="#999999"
                    hidesWhenStopped={true}
                    size="small"
                  />
                </View>
              </View>
            </View>
            <Modal
              animated={true}
              animationType="slide"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              visible={false}
            >
              <RCTSafeAreaView
                emulateUnlessSupported={true}
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCorrect={false}
                    autoFocus={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onClose={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter country name"
                    placeholderTextColor="#aaa"
                    rejectResponderTermination={true}
                    renderFilter={[Function]}
                    style={
                      Array [
                        Object {
                          "height": 48,
                          "width": "70%",
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                      ]
                    }
                    testID="text-input-country-filter"
                    underlineColorAndroid="transparent"
                    value=""
                    withCloseButton={false}
                    withFilter={true}
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                      },
                    ]
                  }
                >
                  <RCTScrollView
                    ItemSeparatorComponent={[Function]}
                    automaticallyAdjustContentInsets={false}
                    data={Array []}
                    disableVirtualization={false}
                    getItem={[Function]}
                    getItemCount={[Function]}
                    getItemLayout={[Function]}
                    horizontal={false}
                    initialNumToRender={14}
                    keyExtractor={[Function]}
                    keyboardShouldPersistTaps="handled"
                    maxToRenderPerBatch={10}
                    numColumns={1}
                    onContentSizeChange={[Function]}
                    onEndReachedThreshold={2}
                    onLayout={[Function]}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    onScrollEndDrag={[Function]}
                    onScrollToIndexFailed={[Function]}
                    removeClippedSubviews={false}
                    renderItem={[Function]}
                    scrollEventThrottle={1}
                    stickyHeaderIndices={Array []}
                    testID="list-countries"
                    updateCellsBatchingPeriod={50}
                    viewabilityConfigCallbackPairs={Array []}
                    windowSize={21}
                  >
                    <View />
                  </RCTScrollView>
                </View>
              </RCTSafeAreaView>
            </Modal>
            <Text
              style={
                Object {
                  "color": "#012D6C",
                  "fontSize": 18,
                  "paddingTop": 5,
                }
              }
            >
              Nepal
            </Text>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "red",
                "flex": 0.6,
                "justifyContent": "flex-start",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Mobile Number"
              placeholderTextColor="#012D6C"
              rejectResponderTermination={true}
              required={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "color": "#012D6C",
                  "fontSize": 18,
                  "height": 45,
                  "padding": 10,
                }
              }
              type="string"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View>
        <View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onChangeText={[Function]}
            placeholder="Brand of the Phone"
            placeholderTextColor="#012D6C"
            rejectResponderTermination={true}
            required={true}
            style={
              Object {
                "backgroundColor": "white",
                "color": "#012D6C",
                "flex": 10,
                "fontSize": 18,
                "height": 45,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 25,
                "padding": 10,
              }
            }
            type="string"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 5,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FF0000",
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Save your Detail
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#012D6C",
            "fontSize": 25,
            "marginBottom": 10,
            "marginTop": 15,
            "textAlign": "center",
          }
        }
      >
         Survey Data
      </Text>
      <View
        style={
          Array [
            undefined,
            Object {
              "borderBottomWidth": 4,
              "borderColor": "#c8e1ff",
              "borderLeftWidth": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#c8e1ff",
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#c8e1ff",
                  "borderRightWidth": 4,
                  "borderTopWidth": 4,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Name
                 
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#c8e1ff",
                  "borderRightWidth": 4,
                  "borderTopWidth": 4,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Country
                 
              </Text>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  sort
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#c8e1ff",
                  "borderRightWidth": 4,
                  "borderTopWidth": 4,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Mobile No.
                 
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#c8e1ff",
                  "borderRightWidth": 4,
                  "borderTopWidth": 4,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Brand
                 
              </Text>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  sort
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              0,
              0,
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
